packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -435,7 +435,7 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
       case TimeOfDayFormat.HH_colon_mm:
         return _formatTwoDigitZeroPad(timeOfDay.hour);
       default:
-        throw AssertionError('$runtimeType does not support $format.');
```

Comment:
This one is ok because it's part of an assertion.

Is there some way to make the lint recognize whether the call is within an `assert` or `throw`?

Reply:
The [lint does test](https://github.com/a14n/linter/blob/avoid_runtimeType_toString/lib/src/rules/avoid_runtimeType_toString.dart) if it's inside an assertion. I don't see assert in this case.

Reply:
Ah yes, I was imprecise.  It should still be ok if it's part of a `throw`. At that point, we don't really care about performance as much anymore and having the real type information would be helpful.

Reply:
I reverted this change.